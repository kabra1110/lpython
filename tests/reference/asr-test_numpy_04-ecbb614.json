{
    "basename": "asr-test_numpy_04-ecbb614",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/test_numpy_04.py",
    "infile_hash": "0fbd0a4280966dd9a917d3a005887954325fb1ea5468fd28dcae930f",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-test_numpy_04-ecbb614.stdout",
    "stdout_hash": "ccc6b3e8a19f7b5aff49cef591cc45883a979fc7a1f3b7665496b242",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}