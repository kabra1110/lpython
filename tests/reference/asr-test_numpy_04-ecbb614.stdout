(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        204
                        {
                            _lpython_main_program:
                                (Function
                                    (SymbolTable
                                        203
                                        {
                                            
                                        })
                                    _lpython_main_program
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [check]
                                    []
                                    [(SubroutineCall
                                        204 check
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            check:
                                (Function
                                    (SymbolTable
                                        201
                                        {
                                            
                                        })
                                    check
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [test_array_01
                                    test_array_02]
                                    []
                                    [(SubroutineCall
                                        204 test_array_01
                                        ()
                                        []
                                        ()
                                    )
                                    (SubroutineCall
                                        204 test_array_02
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            test_array_01:
                                (Function
                                    (SymbolTable
                                        199
                                        {
                                            eps:
                                                (Variable
                                                    199
                                                    eps
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    199
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 3 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    test_array_01
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(=
                                        (Var 199 x)
                                        (ArrayConstant
                                            [(RealConstant
                                                1.000000
                                                (Real 8)
                                            )
                                            (RealConstant
                                                2.000000
                                                (Real 8)
                                            )
                                            (RealConstant
                                                3.000000
                                                (Real 8)
                                            )]
                                            (Array
                                                (Real 8)
                                                [((IntegerConstant 0 (Integer 4))
                                                (IntegerConstant 3 (Integer 4)))]
                                                PointerToDataArray
                                            )
                                            RowMajor
                                        )
                                        ()
                                    )
                                    (=
                                        (Var 199 eps)
                                        (RealConstant
                                            0.000000
                                            (Real 8)
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (IntrinsicFunction
                                                Abs
                                                [(RealBinOp
                                                    (ArrayItem
                                                        (Var 199 x)
                                                        [(()
                                                        (IntegerConstant 0 (Integer 4))
                                                        ())]
                                                        (Real 8)
                                                        RowMajor
                                                        ()
                                                    )
                                                    Sub
                                                    (RealConstant
                                                        1.000000
                                                        (Real 8)
                                                    )
                                                    (Real 8)
                                                    ()
                                                )]
                                                0
                                                (Real 8)
                                                ()
                                            )
                                            Lt
                                            (Var 199 eps)
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (IntrinsicFunction
                                                Abs
                                                [(RealBinOp
                                                    (ArrayItem
                                                        (Var 199 x)
                                                        [(()
                                                        (IntegerConstant 1 (Integer 4))
                                                        ())]
                                                        (Real 8)
                                                        RowMajor
                                                        ()
                                                    )
                                                    Sub
                                                    (RealConstant
                                                        2.000000
                                                        (Real 8)
                                                    )
                                                    (Real 8)
                                                    ()
                                                )]
                                                0
                                                (Real 8)
                                                ()
                                            )
                                            Lt
                                            (Var 199 eps)
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (IntrinsicFunction
                                                Abs
                                                [(RealBinOp
                                                    (ArrayItem
                                                        (Var 199 x)
                                                        [(()
                                                        (IntegerConstant 2 (Integer 4))
                                                        ())]
                                                        (Real 8)
                                                        RowMajor
                                                        ()
                                                    )
                                                    Sub
                                                    (RealConstant
                                                        3.000000
                                                        (Real 8)
                                                    )
                                                    (Real 8)
                                                    ()
                                                )]
                                                0
                                                (Real 8)
                                                ()
                                            )
                                            Lt
                                            (Var 199 eps)
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            test_array_02:
                                (Function
                                    (SymbolTable
                                        200
                                        {
                                            eps:
                                                (Variable
                                                    200
                                                    eps
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    200
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 3 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    test_array_02
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(=
                                        (Var 200 x)
                                        (ArrayConstant
                                            [(IntegerConstant 1 (Integer 4))
                                            (IntegerConstant 2 (Integer 4))
                                            (IntegerConstant 3 (Integer 4))]
                                            (Array
                                                (Integer 4)
                                                [((IntegerConstant 0 (Integer 4))
                                                (IntegerConstant 3 (Integer 4)))]
                                                PointerToDataArray
                                            )
                                            RowMajor
                                        )
                                        ()
                                    )
                                    (=
                                        (Var 200 eps)
                                        (RealConstant
                                            0.000000
                                            (Real 8)
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (Cast
                                                (IntrinsicFunction
                                                    Abs
                                                    [(IntegerBinOp
                                                        (ArrayItem
                                                            (Var 200 x)
                                                            [(()
                                                            (IntegerConstant 0 (Integer 4))
                                                            ())]
                                                            (Integer 4)
                                                            RowMajor
                                                            ()
                                                        )
                                                        Sub
                                                        (IntegerConstant 1 (Integer 4))
                                                        (Integer 4)
                                                        ()
                                                    )]
                                                    0
                                                    (Integer 4)
                                                    ()
                                                )
                                                IntegerToReal
                                                (Real 8)
                                                ()
                                            )
                                            Lt
                                            (Var 200 eps)
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (Cast
                                                (IntrinsicFunction
                                                    Abs
                                                    [(IntegerBinOp
                                                        (ArrayItem
                                                            (Var 200 x)
                                                            [(()
                                                            (IntegerConstant 1 (Integer 4))
                                                            ())]
                                                            (Integer 4)
                                                            RowMajor
                                                            ()
                                                        )
                                                        Sub
                                                        (IntegerConstant 2 (Integer 4))
                                                        (Integer 4)
                                                        ()
                                                    )]
                                                    0
                                                    (Integer 4)
                                                    ()
                                                )
                                                IntegerToReal
                                                (Real 8)
                                                ()
                                            )
                                            Lt
                                            (Var 200 eps)
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (Cast
                                                (IntrinsicFunction
                                                    Abs
                                                    [(IntegerBinOp
                                                        (ArrayItem
                                                            (Var 200 x)
                                                            [(()
                                                            (IntegerConstant 2 (Integer 4))
                                                            ())]
                                                            (Integer 4)
                                                            RowMajor
                                                            ()
                                                        )
                                                        Sub
                                                        (IntegerConstant 3 (Integer 4))
                                                        (Integer 4)
                                                        ()
                                                    )]
                                                    0
                                                    (Integer 4)
                                                    ()
                                                )
                                                IntegerToReal
                                                (Real 8)
                                                ()
                                            )
                                            Lt
                                            (Var 200 eps)
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    _global_symbols
                    []
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        202
                        {
                            _lpython_main_program:
                                (ExternalSymbol
                                    202
                                    _lpython_main_program
                                    204 _lpython_main_program
                                    _global_symbols
                                    []
                                    _lpython_main_program
                                    Public
                                )
                        })
                    main_program
                    [_global_symbols]
                    [(SubroutineCall
                        202 _lpython_main_program
                        ()
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
